src/components/pages/SignIn.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {useState, FormEvent, useEffect} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+// import { Link } from 'react-router-dom';
+
+import Avatar from '@material-ui/core/Avatar';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Link from '@material-ui/core/Link';
+import Grid from '@material-ui/core/Grid';
+import Box from '@material-ui/core/Box';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import {makeStyles} from '@material-ui/core/styles';
+import Container from '@material-ui/core/Container';
+
+import UIInput from '../elements/Input';
+import UIButton from '../elements/Button';
+import Message from '../elements/Message';
+import {signin, setError} from '../../actions/authActions';
+import {RootState} from '../../store';
+
+const Copyright = () => {
+  return (
+    <Typography variant="body2" color="textSecondary" align="center">
+      {'Copyright © '}
+      <Link color="inherit" href="https://material-ui.com/">
+        AutoRack
+      </Link>{' '}
+      {new Date().getFullYear()}
+      {'.'}
+    </Typography>
+  );
+};
+
+const useStyles = makeStyles((theme) => ({
+  paper: {
+    marginTop: theme.spacing(16),
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+  },
+  avatar: {
+    margin: theme.spacing(1),
+    backgroundColor: theme.palette.secondary.main,
+  },
+  form: {
+    width: '100%', // Fix IE 11 issue.
+    marginTop: theme.spacing(1),
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+}));
+
+export const SignIn = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [loading, setLoading] = useState(false);
+  const dispatch = useDispatch();
+  const {error} = useSelector((state: RootState) => state.auth);
+
+  useEffect(() => {
+    return () => {
+      if (error) {
+        dispatch(setError(''));
+      }
+    };
+  }, [error, dispatch]);
+
+  const submitHandler = (e: FormEvent) => {
+    e.preventDefault();
+    if (error) {
+      dispatch(setError(''));
+    }
+    setLoading(true);
+    dispatch(signin({email, password}, () => setLoading(false)));
+  };
+
+  const classes = useStyles();
+
+  return (
+    <Container component="main" maxWidth="xs">
+      <CssBaseline />
+      <div className={classes.paper}>
+        <Avatar className={classes.avatar}>
+          <LockOutlinedIcon />
+        </Avatar>
+        <Typography component="h1" variant="h5">
+            Sign in
+        </Typography>
+        <form className={classes.form} onSubmit={submitHandler}>
```

Comment:
Extract duplicated code to be single components and re-use it

Reply:
So make an element component for the Typography type?

Reply:
Usually (and what's my work in progress) is to create a generic form to be used. You'll be able to see that once I've pushed it